### PR TITLE
Clarify memory management of returned petsc4py objects.

### PR DESCRIPTION
**Pull request by jhale**
Explicit instructions for handling petsc4py objects memory management.

### CODE REVIEW

**Comment by jorgensd** [comment]
Some context: https://gitlab.com/petsc/petsc/-/issues/1309

**Comment by jhale** [comment]
These additions are correct at least according to my understanding of the new petsc4py memory management model.

File: python/dolfinx/fem/petsc.py [hunk 1]
```diff
@@ -111,6 +111,10 @@ def create_vector(L: Form) -> PETSc.Vec:
 def create_vector_block(L: list[Form]) -> PETSc.Vec:
     """Create a PETSc vector (blocked) that is compatible with a list of linear forms.
 
+    Note:
```
Comment (garth-wells) [inline python/dolfinx/fem/petsc.py:114]: Suggested text:
```
Note:
    Due to subtle issues in the interaction between petsc4py memory management 
    and the Python garbage collector, it is recommended that the method ``PETSc.Vec.destroy()`` 
    is called on the returned object once the object is no longer required. Note that 
    ``PETSc.Vec.destroy()`` is collective over the object's MPI communicator.  
```

**Review by garth-wells — CHANGES_REQUESTED** [review]
I've added suggest text that is technically accurate.

**Review by garth-wells — APPROVED** [review]

**Pull request merged**